packages/devtools/test/integration_tests/server_test.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@TestOn('vm')
+import 'dart:io';
+
+import 'package:test/test.dart';
+import 'package:vm_service_lib/vm_service_lib.dart';
+
+import '../support/cli_test_driver.dart';
+import '../support/devtools_server_driver.dart';
+import 'integration.dart';
+
+void main() {
+  CliAppFixture appFixture;
+  DevToolsServerDriver server;
+
+  setUp(() async {
+    // Build the app, as the server can't start without the build output.
+    await WebdevFixture.build(verbose: true);
+
+    // The packages folder needs to be renamed to `pack` for the server to work.
+    if (await Directory('build/pack').exists()) {
+      await Directory('build/pack').delete(recursive: true);
+    }
+    await Directory('build/packages').rename('build/pack');
+
+    // Start the command-line server.
+    server = await DevToolsServerDriver.create();
+
+    // Start a test app we can use to connect to.
+    appFixture = await CliAppFixture.create('test/fixtures/empty_app.dart');
+  });
+
+  tearDown(() async {
+    server?.kill();
+    await appFixture?.teardown();
+  });
+
+  test('registers service', () async {
+    // Track services as they're registered.
+    final registeredServices = <String>{};
+    // TODO(dantup): Remove this loop and just use
+    // appFixture.serviceConnection.onServiceEvent
+    // directly once the VM Service makes it to stable.
```

Comment:
Seems a bit odd that we are connecting to _Service even on the latest VM. Perhaps add a common helper to the non-test code that gives the name 'Service' vs '_Service' that should be used everywhere.